src/Context/TodosContext.tsx

```diff
@@ -0,0 +1,157 @@
+  handleDeleteCompleted: () => { },
+  handleStatus: () => { },
+  handleAllCompleted: () => { },
+  handleUpdateTodo: () => { },
+  handleDeleteTodo: () => { },
+  handleApiTodos: () => { },
+  handleErrorMessage: () => { },
+});
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState(Status.All);
+  const [errorMessage, setErrorMessage] = useState(Error.None);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+
+  const handleErrorMessage = (error: Error) => {
+    setErrorMessage(error);
+  };
+
+  const handleApiTodos = (response: Todo[]) => {
+    setTodos(response);
+  };
+
+  const handleDeleteCompleted = () => {
+    const deleteAllCompleted = todos.filter(todo => !todo.completed);
+
+    setTodos(deleteAllCompleted);
+  };
+
+  const handleCompleted = (todoId: number) => {
+    setTodos(todos.map(todo => (todo.id === todoId
+      ? { ...todo, completed: !todo.completed }
+      : todo)));
+  };
+
+  const handleAllCompleted = () => {
+    const statusCompleted = todos.some(todo => !todo.completed);
+
+    if (statusCompleted) {
+      setTodos(todos.map(todo => (todo.completed === false
+        ? { ...todo, completed: !todo.completed }
+        : todo)));
+    } else {
+      setTodos(todos.map(
+        todo => ({ ...todo, completed: !todo.completed }),
+      ));
+    }
+  };
+
+  const handleTodo = ({ title, userId, completed }: Omit<Todo, 'id'>) => {
+    setTempTodo({
+      id: 0,
+      userId,
+      title,
+      completed,
+    });
+
+    return todoService.createTodos({ title, userId, completed })
+      .then(newTodo => {
+        setTodos(
+          [
+            ...todos,
+            newTodo,
+          ],
+        );
+      })
+      .finally(() => {
+        setTempTodo(null);
+      });
+  };
+
+  const handleUpdateTodo = (changeId: number, updateTitle: string) => {
+    setTodos(todos.map(todo => (todo.id === changeId
+      ? { ...todo, title: updateTitle }
+      : todo)));
+  };
+
+  const handleDeleteTodo = (deleteId: number) => {
+    setLoadingIds(state => [
+      ...state,
+      deleteId,
+    ]);
+
+    todoService.deleteTodos(`/todos/${deleteId}`)
+      .then(() => {
+        const filteredTodos = todos.filter(todo => todo.id !== deleteId);
+
+        setTodos(filteredTodos);
+      })
+      .finally(() => {
+        setLoadingIds([]);
+      });
+  };
+
+  const handleStatus = (newStatus: Status) => {
+    setStatus(newStatus);
+  };
+
+  const value = useMemo(() => ({
+    todos,
+    status,
+    errorMessage,
+    tempTodo,
+    loadingIds,
+    handleTodo,
+    handleCompleted,
+    handleDeleteCompleted,
+    handleStatus,
+    handleAllCompleted,
+    handleUpdateTodo,
+    handleDeleteTodo,
+    handleApiTodos,
+    handleErrorMessage,
+  }), [todos, status, errorMessage, tempTodo, loadingIds]);
```

Comment:
fix linter error here

Reply:
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/108538921/2890238e-712b-4062-ba23-3995a7e53a07)